openfisca_core/reforms.py

```diff
@@ -1,34 +1,45 @@
 # -*- coding: utf-8 -*-
 
 import copy
-import collections
-import json
 
-from biryani.strings import deep_encode
-
-from . import conv, legislations
+from . import legislations
 from .taxbenefitsystems import TaxBenefitSystem
 
 
-def compose_reforms(reforms, tax_benefit_system):
-    """
-    Compose reforms: the first reform is built with the given base tax-benefit system,
-    then each one is built with the previous one as the baseline.
-    """
-    def compose_reforms_reducer(memo, reform):
-        reformed_tbs = reform(memo)
-        return reformed_tbs
-    final_tbs = reduce(compose_reforms_reducer, reforms, tax_benefit_system)
-    return final_tbs
-
-
 class Reform(TaxBenefitSystem):
+    """A modified TaxBenefitSystem
+
+    A reformed TaxBenefitSystem must subclass `Reform` and implement a method `apply()`. Such a function can add or replace variables and call `self.modify_legislation()` to modify the parameters of the legislation.
```

Comment:
>  In OpenFisca, a reform is a modified TaxBenefitSystem.
 It can add or replace variables and call self.modify_legislation()` to modify the parameters of the legislation.
All reforms must subclass `Reform` and implement a `apply()` method .